### PR TITLE
[MoE] Add nemotron_h fast loading support

### PR DESCRIPTION
**Pull request by kylesayrs**
SUMMARY:
Add `nemotron_h` support to the MoE linearization / conversion-mapping machinery so that NemotronH models (e.g. Nemotron 3 Ultra) can be loaded and saved as per-expert 2D tensors. The architecture was already registered in `ARCH_TO_IMPORT_PATHS`; this adds the missing pieces:

- Registers a 2D conversion mapping for `nemotron_h` in `ARCH_TO_2D_MAPPINGS`. NemotronH experts are **non-gated** (only `up_proj` + `down_proj`, no `gate_proj`), and the fused checkpoint stores them as 3D `mixer.experts.{up,down}_proj` tensors, so the mapping drops those fused 3D targets and renames to per-expert 2D.
- Maps `MoEConfig.hidden_dim` to `config.moe_latent_size` for `nemotron_h` when a latent projection is used, falling back to `hidden_size` otherwise. (Nemotron 3 Ultra uses `moe_latent_size=2048` rather than `hidden_size=8192` as the expert input/output dim.)
- Adds unit tests covering the `nemotron_h` mapping and the `hidden_dim`/`moe_latent_size` behavior.
- Adds a REAP + GPTQ `NVFP4` example for `nvidia/NVIDIA-Nemotron-3-Ultra-550B-A55B-BF16`.

Note: this deliberately extracts only the `nemotron_h`-related parts from the `support nemotron` commit (6228f95). Unrelated changes from that commit — the `kimi_k3_example.py` edits and the unused-`CompressionLogger`-import removal in `gptq/base.py` — are excluded here.

TEST PLAN:
- `pytest tests/llmcompressor/modeling/test_conversion_mappings.py tests/llmcompressor/modeling/test_linearize.py` — all pass, including the new `test_nemotron_h_*` and `test_moe_config_nemotron_h_hidden_dim` tests and the existing `test_linearize_moe[nemotron_h]`.
- `ruff check` clean on all touched files.
- Verified `MoEConfig.from_config` returns `hidden_dim=2048` for a latent NemotronH config and `hidden_dim=32` for a non-latent one.

### CODE REVIEW

File: examples/reap_expert_pruning/nemotron_example.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# NOTE: to use a custom dataset, see examples/custom_dataset_example.py
```
Comment (kylesayrs) [inline examples/reap_expert_pruning/nemotron_example.py]: I'll remove this

File: src/llmcompressor/modeling/moe/conversion_mappings.py [hunk 2]
```diff
@@ -214,6 +214,21 @@
             ),
         ],
     ),
+    # NemotronH experts are non-gated and the fused checkpoint stores them as
+    # 3D `mixer.experts.{up,down}_proj` tensors
```
Comment (kylesayrs) [inline src/llmcompressor/modeling/moe/conversion_mappings.py]: There is no 3d checkpoint

File: src/llmcompressor/modeling/moe/helpers.py [hunk 3]
```diff
@@ -153,6 +153,11 @@ def from_config(cls, config: PreTrainedConfig):
                 ret.hidden_act = "sigmoid"
             case "lfm2_moe":
                 ret.hidden_act = "silu"
+            case "nemotron_h":
```
Comment (kylesayrs) [inline src/llmcompressor/modeling/moe/helpers.py]: This should just be added to the fallback list

File: tests/llmcompressor/modeling/test_conversion_mappings.py [hunk 4]
```diff
@@ -57,6 +57,45 @@ def test_patch_moe_mappings_removes_targets(restore_mappings):
     assert load_mappings[-len(RENAMINGS) :] == RENAMINGS
 
 
+def test_nemotron_h_has_linearize_load_mappings(restore_mappings):
+    assert has_linearize_load_mappings("nemotron_h")
+
+
+def test_nemotron_h_load_mappings(restore_mappings):
```
Comment (kylesayrs) [inline tests/llmcompressor/modeling/test_conversion_mappings.py]: I'll remove these tests

File: tests/llmcompressor/modeling/test_linearize.py [hunk 5]
```diff
@@ -210,6 +210,26 @@ def test_linearize_moe(model_type):
         assert torch.nn.functional.mse_loss(calib_outputs, true_outputs) < MODULE_MSE
 
 
+def test_moe_config_nemotron_h_hidden_dim():
```
Comment (kylesayrs) [inline tests/llmcompressor/modeling/test_linearize.py]: I'll remove these tests

File: src/llmcompressor/modeling/moe/conversion_mappings.py
Comment (kylesayrs) [inline src/llmcompressor/modeling/moe/conversion_mappings.py:163]: need to add nemotron here

**Review by HDCharles — APPROVED** [review]
